tests/query.json

```diff
@@ -200,7 +209,13 @@
         ["QueryCount", 3, "StringObject", "stringCol ==[c] 'a'"],
         ["QueryCount", 5, "StringObject", "stringCol BEGINSWITH[c] 'A'"],
         ["QueryCount", 4, "StringObject", "stringCol ENDSWITH[c] 'c'"],
-        ["QueryCount", 2, "StringObject", "stringCol CONTAINS[c] 'B'"]
+        ["QueryCount", 2, "StringObject", "stringCol CONTAINS[c] 'B'"],
+        ["QueryCount", 2, "StringObject", "stringCol LIKE[c] '*B?'"],
+        ["QueryCount", 0, "StringObject", "stringCol LIKE 1"],
+        ["QueryCount", 0, "StringObject", "stringCol LIKE LIKE"],
+        ["QueryCount", 0, "StringObject", "stringCol LIKE[A] '*BC'"],
```

Comment:
`[c]` / `[C]` is the only modifier that's accepted based on https://github.com/realm/realm-object-store/blob/86138ce4e55ca04e975c1165d409f02dcb729b6b/src/parser/parser.cpp#L73

I'd expect `stringCol LIKE 1` to _parse_ successfully but be rejected when we attempt to build a query from it due to the type mismatch.